gensim/models/fasttext.py

```diff
@@ -850,25 +850,15 @@ def load(cls, *args, **kwargs):
             Save :class:`~gensim.models.fasttext.FastText` model.
 
         """
-        model = super(FastText, cls).load(*args, rethrow=True, **kwargs)
-
-        if not hasattr(model.wv, 'vectors_vocab_lockf') and hasattr(model.wv, 'vectors_vocab'):
-            # TODO: try trainables-location
-            model.wv.vectors_vocab_lockf = ones(1, dtype=REAL)
-        if not hasattr(model, 'vectors_ngrams_lockf') and hasattr(model.wv, 'vectors_ngrams'):
-            # TODO: try trainables-location
-            model.wv.vectors_ngrams_lockf = ones(1, dtype=REAL)
-        # fixup mistakenly overdimensioned gensim-3.x lockf arrays
-        if len(model.wv.vectors_vocab_lockf.shape) > 1:
-            model.wv.vectors_vocab_lockf = ones(1, dtype=REAL)
-        if len(model.wv.vectors_ngrams_lockf.shape) > 1:
-            model.wv.vectors_ngrams_lockf = ones(1, dtype=REAL)
-        if hasattr(model, 'bucket'):
-            del model.bucket  # should only exist in one place: the wv subcomponent
-        if not hasattr(model.wv, 'buckets_word') or not model.wv.buckets_word:
-            model.wv.recalc_char_ngram_buckets()
+        return super(FastText, cls).load(*args, rethrow=True, **kwargs)
```

Comment:
Do we really need such stub methods that only calls the same superclass method with the same arguments? That's already the default which occurs if no method is present. By my understanding, doc-comment tools like Sphinx will, in their current versions, already propagate superclass API docs down to subclasses.

The only thing that's varying is the comment, and while it expresses a different expected-type from the superclass, in practice that doc may be misleading: I **think** (but have not recently checked) that these `SaveLoad` `.load()` methods can return objects that **may not be** what the caller expects. They return **the class that's in the file**, not the class-that-`.load()`-was-called-on.

If so, it might be a worthwhile short-term step as soon as 4.0.0 – for limiting the risk of confusion & requirement for redundant/caveat-filled docs – to **deprecate the practice of calling *SpecificClass*.load(filename) entirely**, despite its common appearance in previously-idiomatic gensim example code. Instead, either (1) call it only on class `SaveLoad` itself, to express that the only expectation for the returned type is that it's a `SaveLoad` subclass; (2) promote load functionality to model-specific top-level functions in each relevant model – a bit more like the `load_facebook_model()` function for loading Facebook-FasttText-native-format models – which might themselves do some type-checking, so any docs which imply they return a certain type are true; (3) just make one `utils.py` generic `load()`, perhaps with an optional class-enforcement parameter, and encourage its use. 

Despite the effort involved in making such changes, they could minimize duplicated code/comments & avoid some unintuitive gotchas in the current `SaveLoad` approach.

Reply:
`SpecificClass.load()` is there to allow custom logic overrides, IIRC. To clean up backward compatibility from loaded objects etc. So exactly your item `(2) promote load functionality to model-specific top-level functions in each relevant model`?

Note I didn't make any changes to this code in this PR, the stubs must be coming from elsewhere (older).

Reply:
> unintuitive gotchas in the current SaveLoad approach

What are they? I was under the impression `save`/`load` was straightforward. Were users complaining?

Reply:
Moved to a separate ticket, so this doesn't get lost: #2957.